app/src/main/java/com/yacgroup/yacguide/statistics/StatisticGenerator.kt

```diff
@@ -64,15 +67,38 @@ class StatisticGenerator(private val _ctx: Context) {
     }
 
     fun generateMostFrequentPartnersStatistic(): Statistic {
+        val statsTitle = _ctx.getString(R.string.ascends_per_partner)
         val ascendCountsPerPartner = StatisticUtils.getAscendCountsPerPartner(_db.getAscendsBelowStyleId(AscendStyle.eBOTCHED.id))
-
         return Statistic(
-            _ctx.getString(R.string.ascends_per_partner),
-            listOf(_ctx.getString(R.string.ascends_per_partner)), // not used because only single-stack
+            statsTitle,
+            listOf(statsTitle), // not used because only single-stack
             _db.getPartners().sortedBy { ascendCountsPerPartner.get(it.id, 0) }.associate { partner ->
                 partner.name.orEmpty() to listOf(ascendCountsPerPartner.get(partner.id))
             }
         )
     }
 
+    fun generateNewlyCollectedRockCountsStatistic(): Statistic {
+        val considerOnlyLeads = _customSettings.getBoolean(_ctx.getString(R.string.pref_key_count_only_leads),
+            _ctx.resources.getBoolean(R.bool.pref_default_count_only_leads))
+        val startStyleId = if (considerOnlyLeads)
+                AscendStyle.eSOLO.id
+            else
+                AscendStyle.eUNKNOWN.id
+        val endStyleId = if (considerOnlyLeads)
+                AscendStyle.eHOCHGESCHLEUDERT.id
+            else
+                AscendStyle.eHINTERHERGEHAMPELT.id
+        val rockCounter = RockCounter(RockCounterConfig.generate(_ctx, _customSettings))
+        val statsTitle = _ctx.getString(R.string.newly_collected_rocks_per_year)
+        return Statistic(
+            statsTitle,
+            listOf(statsTitle),
+            _db.getNewlyAscendedRockCountsPerYear(startStyleId, endStyleId, rockCounter.getConsideredRockTypes(), rockCounter.getExcludedRockStates()).associate { rockCount ->
```

Comment:
I see enums as something where the programmer does not need to care about the value behind. With this `startStyleId`/`endStyleId` approach this is not true anymore. I think, we also have other places in the code like this. Maybe we should consider to handle names groups of styles which are handled on a central place.

Reply:
I agree, some refactoring in the future might be neat.